src/lib/cli.ts

```diff
@@ -44,101 +42,105 @@ exports.copy = function (src, dst) {
 		});
 	}
 	else {
-		var data = fs.readFileSync(src);
+		const data = fs.readFileSync(src);
 		fs.writeFileSync(dst, data);
 	}
-};
+}
 
 /**
  * Ensures a value is part of an enum
  */
-exports.enumArg = function (choices, val) {
+export function enumArg(choices: any[], val: any): any {
 	if (choices.indexOf(val) === -1) {
 		exports.die('error: expected "' + val + '" to be one of {' + choices.join(', ') + '}');
 	}
 	return val;
-};
+}
 
 /**
  * Get the exit code corresponding to a signal name
  */
-exports.exitCodeForSignal = function (signalName) {
-	return 128 + (os.constants.signals[signalName] || 0);
-};
+export function exitCodeForSignal(signalName: string): number {
+	return 128 + ((<any> os.constants.signals)[signalName] || 0);
+}
 
 /**
  * Prints a message to the console
  */
-exports.print = function () {
-	if (arguments.length === 1 && Array.isArray(arguments[0])) {
-		console.log(format(arguments[0].join('\n')));
+export function print(...args: any[]): void {
+	if (args.length === 1 && Array.isArray(args[0])) {
+		console.log(format((<any[]> args[0]).join('\n')));
 	}
 	else {
-		console.log(format.apply(null, arguments));
+		console.log(format.apply(null, args));
 	}
-};
+}
 
 /**
  * Logs an error message and exits
  */
-exports.die = function () {
+export let die: (...args: any[]) => void;
+
+die = function (...args: any[]): void {
 	console.error();
 
-	if (arguments.length === 1 && Array.isArray(arguments[0])) {
-		console.error(format(arguments[0].join('\n')));
+	if (args.length === 1 && Array.isArray(args[0])) {
+		console.error(format(args[0].join('\n')));
 	}
 	else {
-		console.error(format.apply(null, arguments));
+		console.error(format.apply(null, args));
 	}
 
 	console.error();
 	process.exit(1);
 };
 
+export function _setDieMethod(method: (...args: any[]) => void) {
+	die = method;
+}
+
 /**
  * Returns a function that will print messages to stderr if the verbose parameter is true. The function will do nothing
  * if verbose is false.
  */
-exports.getLogger = function (verbose) {
+export function getLogger(verbose?: boolean): Function {
 	if (verbose) {
-		var util = require('util');
 		return function () {
 			process.stderr.write('>> ' + util.format.apply(util, arguments) + '\n');
```

Comment:
This would read better w/ rest parameter:

```ts
return function(...args: any[]) {
    process.stderr.write(`>> ${ util.format(... args) }\n`);
};